src/main/java/eu/ai4work/sws/config/FuzzyInferenceSystemInitializer.java

```diff
@@ -34,22 +35,22 @@ public FIS initializeFuzzyInferenceSystem() throws FileNotFoundException, Invali
         // Try external file system
         File externalFuzzyRuleFile = new File(fclRulesFilePath);
         if (externalFuzzyRuleFile.exists()) {
-            return parseFclFile(externalFuzzyRuleFile.getPath());
+            return parseFclFile(new FileInputStream(externalFuzzyRuleFile), fclRulesFilePath);
         }
 
         // Then try classpath resources
-        URL internalFuzzyRuleResourceUrl = getClass().getClassLoader().getResource(fclRulesFilePath);
-        if (internalFuzzyRuleResourceUrl != null) {
-            return parseFclFile(internalFuzzyRuleResourceUrl.getPath());
+        InputStream internalFuzzyRuleResourceFile = getClass().getClassLoader().getResourceAsStream(fclRulesFilePath);
+        if (internalFuzzyRuleResourceFile != null) {
+            return parseFclFile(internalFuzzyRuleResourceFile, fclRulesFilePath);
         }
 
         throw new FileNotFoundException("Fuzzy Control Language (FCL) file not found: " + fclRulesFilePath);
     }
 
-    private static FIS parseFclFile(String fclRulesFilePath) {
+    private static FIS parseFclFile(InputStream inputStream, String fclRulesFilePath) {
         try {
             logger.debug("Initializing a Fuzzy Inference System (FIS) based on the FCL file: " + fclRulesFilePath);
```

Comment:
Just a minor suggestion to simplify the code: if we move this line to line 33, then we do not need to pass the `String fclRulesFilePath` parameter to this method (as it is only used for logging, not for the actual functionality).

We could also set the log level to "info" here, so that we will always see in the log which file it was trying to parse in case parsing fails (then we can also remove the `fclRulesFilePath` in the exception message in line 55).